dhcp-controller/dhcpserver/api/v1alpha1/ipallocation_types.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// IPAllocationSpec defines the desired state of IPAllocation
+type IPAllocationSpec struct {
+	// Range is a string that represents an IP address
+	Range string `json:"range"`
+	// Allocations is the set of allocated IPs for the given range. Its` indices are a direct mapping to the
+	// IP with the same index/offset for the allocation's range.
+	Allocations map[string]IPAllocationOwner `json:"allocations"`
+	// EpochExpiry is the epoch time when the IP was set to expire in the leasefile
+	EpochExpiry string `json:"epochexpiry"`
+}
+
+// ParseCIDR formats the Range of the IPAllocation
+func (i IPAllocation) ParseCIDR() (net.IP, *net.IPNet, error) {
+	return net.ParseCIDR(i.Spec.Range)
+}
+
+// IPAllocationOwner represents metadata about the pod/container owner of a specific IP
+type IPAllocationOwner struct {
+	MacId  string `json:"id,omitempty"`
```

Comment:
What is MacId? is it mac address? If so can we rename it to MacAddr. If not, is there a reason to use a different UUID for MAC when macs should be unique?

Reply:
MacId was a misnomer. Changed to MacAddr.